components/user_settings/advanced/join_leave_section/join_leave_section.jsx

```diff
@@ -0,0 +1,130 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React from 'react';
+import PropTypes from 'prop-types';
+import {FormattedMessage} from 'react-intl';
+
+import {Preferences} from 'mattermost-redux/constants';
+
+import SettingItemMax from 'components/setting_item_max.jsx';
+import SettingItemMin from 'components/setting_item_min.jsx';
+
+import {AdvancedSections} from 'utils/constants.jsx';
+
+export default class JoinLeaveSection extends React.PureComponent {
+    static propTypes = {
+        activeSection: PropTypes.string,
+        currentUserId: PropTypes.string.isRequired,
+        joinLeave: PropTypes.string,
+        onUpdateSection: PropTypes.func.isRequired,
+        prevActiveSection: PropTypes.string,
+        renderOnOffLabel: PropTypes.func.isRequired,
+        actions: PropTypes.shape({
+            savePreferences: PropTypes.func.isRequired,
+        }).isRequired,
+    }
+
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            joinLeaveState: props.joinLeave,
+        };
+    }
+
+    handleOnChange = (e) => {
+        const value = e.currentTarget.value;
+
+        this.setState({joinLeaveState: value});
+    }
+
+    handleSubmit = () => {
+        const {actions, currentUserId, onUpdateSection} = this.props;
+        const joinLeavePreference = {category: Preferences.CATEGORY_ADVANCED_SETTINGS, user_id: currentUserId, name: 'join_leave', value: this.state.joinLeaveState};
```

Comment:
please make a constant for `'join_leave'`

Reply:
Oh right, now updated. Thanks